src/m365/spfx/commands/project/project-externalize/rules/DynamicRule.ts

```diff
@@ -17,50 +17,54 @@ export class DynamicRule extends BasicDependencyRule {
   private restrictedNamespaces = ['@types/', '@microsoft/'];
   private fileVariationSuffixes = ['.min', '.bundle', '-min', '.bundle.min'];
 
-  public visit(project: Project): Promise<VisitationResult> {
-    if (!project.packageJson ||
-      !project.packageJson.dependencies) {
-      return Promise.resolve({ entries: [], suggestions: [] });
+  public async visit(project: Project): Promise<VisitationResult> {
+    if (!project.packageJson || !project.packageJson.dependencies) {
+      return { entries: [], suggestions: [] };
     }
 
     const validPackageNames: string[] = Object.getOwnPropertyNames(project.packageJson.dependencies)
       .filter(x => this.restrictedNamespaces.map(y => x.indexOf(y) === -1).reduce((y, z) => y && z))
       .filter(x => this.restrictedModules.indexOf(x) === -1);
 
-    return Promise
-      .all(validPackageNames.map((x) => this.getExternalEntryForPackage(x, project)))
-      .then((res: (ExternalizeEntry | undefined)[]) => {
-        return {
-          entries: res
-            .filter(x => x !== undefined)
-            .map(x => x as ExternalizeEntry),
-          suggestions: []
-        };
-      });
+    const res: (ExternalizeEntry | undefined)[] = await Promise.all(
+      validPackageNames.map((x) => this.getExternalEntryForPackage(x, project))
+    );
+
+    const entries = res.filter(x => x !== undefined) as ExternalizeEntry[];
+
+    return {
+      entries,
+      suggestions: []
+    };
   }
 
-  private getExternalEntryForPackage(packageName: string, project: Project): Promise<ExternalizeEntry | undefined> {
+  private async getExternalEntryForPackage(packageName: string, project: Project): Promise<ExternalizeEntry | undefined> {
     const version: string | undefined = project.packageJson!.dependencies![packageName];
     const filesPaths: string[] = this.getFilePath(packageName).map(x => this.cleanFilePath(x));
 
     if (!version || filesPaths.length === 0) {
-      return Promise.resolve(undefined);
+      return undefined;
     }
 
     const filesPathsVariations = filesPaths
       .map(x => this.fileVariationSuffixes.map(y => x.indexOf(y) === -1 ? x.replace('.js', `${y}.js`) : x))
       .reduce((x, y) => [...x, ...y]);
     const pathsAndVariations = [...filesPaths, ...filesPathsVariations];
-    return Promise.all(pathsAndVariations.map(x => this.getExternalEntryForFilePath(x, packageName, version)))
```

Comment:
We appear to have two promise.alls @nicodecleyre. Could you fix this one as well?